roles/create-ai-environment/tasks/main.yaml

```diff
@@ -0,0 +1,289 @@
+  register: runtime_version
+
+- name: Get Framework Name
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_name.txt'
+  register: framework
+
+- name: Get Framework Version
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_version.txt'
+  register: framework_version
+
+- name: Get User Entry Point
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/entry_point.txt'
+  register: entrypoint
+
+- name: Check output folder variable exists
+  stat:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/output_folder.txt'
+  register: stat_ouput_folder_variable
+
+- name: Get output folder variable value
+  when: stat_ouput_folder_variable.stat.exists
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/output_folder.txt'
+  register: output_folder_name
+
+- name: Check output folder exists
+  when: output_folder_name is defined
+  stat:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/{{ output_folder_name.stdout }}'
+  register: stat_ouput_folder
+
+- name: Create output folder if non-exist
+  when: output_folder_name is defined and not stat_ouput_folder.stat.exists
+  file:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/{{ output_folder_name.stdout }}'
+    state: directory
+
+- name: Set training output folder name by file
+  when: output_folder_name is defined
+  set_fact:
+    training_output_folder_name: "{{ output_folder_name.stdout }}"
+
+- name: Generate new output folder if non-exist
+  when: output_folder_name is not defined
+  file:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/training_result'
+    state: directory
+
+- name: Set training output folder name by generation
+  when: output_folder_name is not defined
+  set_fact:
+    training_output_folder_name: "training_result"
+
+- name: Install {{ runtime.stdout }} Runtime
+  when: runtime_name
+  shell:
+    cmd: |
+      set -x
+      set -e
+      runtime_name={{ runtime_name.stdout }}
+      runtime_version={{ runtime_version.stdout }}
+      add-apt-repository -y ppa:deadsnakes/ppa
+      apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F60F4B3D7FA2AF80
+      apt update
+      if [ $runtime_name == 'python' ];then
+        apt install -y $runtime_name$runtime_version
+        apt install -y python2.7 python3.5 python3.6 python3.7
+      fi
+
+      if [ $runtime_name == 'java' ];then
+        apt install -y default-jre default-jdk
+        apt install -y openjdk-7-jre openjdk-8-jdk
+      fi
+
+      if [ $runtime_name == 'c' ];then
+        apt install -y gcc=$runtime_version g++=$runtime_version
+        apt install -y gcc-7 g++-7 gcc-8 g++-8 gcc-9 g++-9
+      fi
+
+      if [ $runtime_name == 'go' ];then
+        apt install -y golang-go=$runtime_version
```

Comment:
I'm not sure have you ever tested to install golang with specfic version, but I try to install 1.10 version:
```shell
$ apt-get install -y golang-go=1.10
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Version '1.10' for 'golang-go' was not found
```
So I guess what you wanted is like:
```
apt-get install -y golang-1.10-go
ln -s ../lib/go-1.10/bin/go /usr/bin/go
```